src/cartesian/Area.tsx

```diff
@@ -104,6 +106,29 @@ type AreaComposedData = ChartOffset & {
   isRange: boolean;
 };
 
+function getLegendItemColor(stroke: string | undefined, fill: string): string {
+  return stroke && stroke !== 'none' ? stroke : fill;
+}
```

Comment:
nit: could this be a common function?

Reply:
As I am untangling the generateCategoricalChart I am finding more and more unique behaviour of each graphical component so I am by default putting everything into separate function. Later once it's done there indeed might be an opportunity to extract some of these as shared code.